NuKeeper/ContainerRegistration.cs

```diff
@@ -38,15 +38,16 @@ private static void Register(Container container)
             container.Register<IUpdateSelection, UpdateSelection>();
             container.Register<IFileSettingsCache, FileSettingsCache>();
 
+            container.RegisterSingleton<ICollaborationFactory,CollaborationFactory>();
+            container.Collection.Register<ISettingsReader>(typeof(GitHubSettingsReader), typeof(AzureDevOpsSettingsReader));
```

Comment:
Would it make sense to generically pull `ISettingsReader` implementations via reflection or do you consider this as overengineering? Potentially this would make future integrations of additional platforms more easy.

Reply:
Yup I like the Idea, can we add it to the list of bonus work rather than complicating this PR?